modules/spanner-instance/variables.tf

```diff
@@ -89,6 +89,11 @@ variable "iam_bindings_additive" {
 variable "instance" {
   description = "Instance attributes."
   type = object({
+    instance_settings = optional(object({
+      default_backup_schedule_type = optional(string)
+      edition                      = optional(string)
+      instance_type                = optional(string)
+    }))
```

Comment:
do you really need a a new nested field here? You could make these 3 attributes top level (and you avoid the `try` mentioned in my previous comment)